sklearn/tests/metadata_routing_common.py

```diff
@@ -153,16 +153,22 @@ def fit(self, X, y, sample_weight="default", metadata="default"):
         record_metadata_not_default(
             self, "fit", sample_weight=sample_weight, metadata=metadata
         )
+        n_features = X.shape[1]
+        self.coef_ = np.ones(n_features)
         return self
 
-    def predict(self, X, sample_weight="default", metadata="default"):
-        pass  # pragma: no cover
+    def predict(self, X, y=None, sample_weight="default", metadata="default"):
+        record_metadata_not_default(
+            self, "predict", sample_weight=sample_weight, metadata=metadata
+        )
+        return np.zeros(shape=(len(X),))
 
-        # when needed, uncomment the implementation
-        # record_metadata_not_default(
-        #     self, "predict", sample_weight=sample_weight, metadata=metadata
-        # )
-        # return np.zeros(shape=(len(X),))
+    def score(self, X, y, sample_weight="default", metadata="default"):
+        self.predict(X, sample_weight="default", metadata="default")
```

Comment:
we don't need to add this, and our existing scorers certainly don't route metadata to predict.

Reply:
No, the existing `score` methods don't route metadata to `predict`. I went through all (I believe) to confirm that. Thus, this line is really not needed here.

For curiosity: What's the problem with `sample_weight` in `predict`? I can see it in any kind of function, except for in predict. I mean, if I modify my data before fitting, I need to modify it using the same pattern before predicting....